pkg/apis/core/validation/validation.go

```diff
@@ -5983,6 +5987,23 @@ func ValidateLoadBalancerStatus(status *core.LoadBalancerStatus, fldPath *field.
 				allErrs = append(allErrs, field.Invalid(idxPath.Child("ip"), ingress.IP, "must be a valid IP address"))
 			}
 		}
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) {
+			if len(ingress.IP) > 0 && ingress.IPMode == nil {
+				allErrs = append(allErrs, field.Required(idxPath.Child("ipMode"), "must be specified when `ip` is set"))
```

Comment:
this is what is failing in the alpha e2e job, so apparently this field is not consistently defaulted. if so, we cannot make it required.

Reply:
True, it was only defaulted on `Service`, completly missed the `Ingress`. Reverted in #96454 and I'll make sure to include ingress in the new one!